crates/burn-dataset/src/dataset/dataframe.rs

```diff
@@ -0,0 +1,228 @@
+use std::io;
+use std::marker::PhantomData;
+
+use polars::frame::DataFrame;
+use polars::frame::row::Row;
+use polars::prelude::*;
+use serde::de::DeserializeOwned;
+use serde_json::{json, to_string, Value};
+
+use crate::Dataset;
+
+pub type Result<T> = core::result::Result<T, DataframeDatasetError>;
+
+/// Custom error type for DataframeDataset type
+#[derive(thiserror::Error, Debug)]
+pub enum DataframeDatasetError {
+    /// IO related error.
+    #[error("IO error: {0}")]
+    Io(#[from] io::Error),
+    /// Serde related error.
+    #[error("Serde error: {0}")]
+    Serde(#[from] rmp_serde::encode::Error),
+    /// Polars related error
+    #[error("Polars error: {0}")]
+    PolarsError(#[from] PolarsError),
+    /// Any other error.
+    #[error("{0}")]
+    Other(&'static str),
+}
+
+impl From<&'static str> for DataframeDatasetError {
+    fn from(s: &'static str) -> Self {
+        DataframeDatasetError::Other(s)
+    }
+}
```

Comment:
I am not sure if these errors are utilized anywhere. Can you make sure you use or remove them?